### PR TITLE
arrayfire, boost-compute and clinfo

### PR DESCRIPTION
**Pull request by glehmann**
several new opencl related libraries and tools.
Originally submitted at https://github.com/Homebrew/homebrew/pull/37928
but moving as suggested

### CODE REVIEW

**Comment by tdsmith** [comment]
Arrayfire's failing on Mountain Lion; I think this is the error:

```
[  5%] Compiling /tmp/arrayfire20150604-9422-sj8r2q/arrayfire/src/backend/opencl/kernel/iops.cl to C++ source
cd /tmp/arrayfire20150604-9422-sj8r2q/arrayfire/src/backend/opencl/kernel && /usr/local/Cellar/cmake/3.2.2/bin/cmake -E make_directory /tmp/arrayfire20150604-9422-sj8r2q/arrayfire/build/src/backend/opencl/kernel_headers
cd /tmp/arrayfire20150604-9422-sj8r2q/arrayfire/src/backend/opencl/kernel && /usr/local/Cellar/cmake/3.2.2/bin/cmake -E echo \#include\ \</tmp/arrayfire20150604-9422-sj8r2q/arrayfire/src/backend/opencl/kernel/iops.hpp\> >>"/tmp/arrayfire20150604-9422-sj8r2q/arrayfire/build/src/backend/opencl/kernel_headers/iops.hpp"
In file included from /tmp/arrayfire20150604-9422-sj8r2q/arrayfire/src/backend/cpu/blas.cpp:12:
In file included from /tmp/arrayfire20150604-9422-sj8r2q/arrayfire/src/api/c/handle.hpp:17:
/tmp/arrayfire20150604-9422-sj8r2q/arrayfire/src/backend/cpu/cast.hpp:37:33: error: variable has incomplete type 'std::complex<float>'
    To eval(std::complex<float> in)
                                ^
/System/Library/Frameworks/vecLib.framework/Headers/vForce.h:31:19: note: forward declaration of 'std::complex<float>'
        template<> class complex<float>;
                         ^
In file included from /tmp/arrayfire20150604-9422-sj8r2q/arrayfire/src/backend/cpu/blas.cpp:12:
In file included from /tmp/arrayfire20150604-9422-sj8r2q/arrayfire/src/api/c/handle.hpp:17:
/tmp/arrayfire20150604-9422-sj8r2q/arrayfire/src/backend/cpu/cast.hpp:46:34: error: variable has incomplete type 'std::complex<double>'
    To eval(std::complex<double> in)
                                 ^
/System/Library/Frameworks/vecLib.framework/Headers/vForce.h:32:19: note: forward declaration of 'std::complex<double>'
        template<> class complex<double>;
                         ^
cd /tmp/arrayfire20150604-9422-sj8r2q/arrayfire/src/backend/opencl/kernel && /tmp/arrayfire20150604-9422-sj8r2q/arrayfire/build/bin2cpp --file iops.cl --namespace opencl --output /tmp/arrayfire20150604-9422-sj8r2q/arrayfire/build/src/backend/opencl/kernel_headers/iops.hpp --name iops_cl --eof 0
/usr/local/Cellar/cmake/3.2.2/bin/cmake -E cmake_progress_report /tmp/arrayfire20150604-9422-sj8r2q/arrayfire/build/CMakeFiles 
[  5%] 2 errors generated.
```

**Comment by glehmann** [comment]
I'll have to find a mountain lion to fix that. It shouldn't be too difficult :-)

File: boost-compute.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
+class BoostCompute < Formula
+  desc "A C++ GPU Computing Library for OpenCL"
+  homepage "https://kylelutz.github.io/compute"
```
Comment (tdsmith) [inline boost-compute.rb]: This homepage 404s.

Comment (tdsmith) [inline boost-compute.rb]: It looks like it's moved here: https://boostorg.github.io/compute/ -- can you update the url as well?

File: arrayfire.rb [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  # build forge separately so we can tell it to use the system freetype
+  resource "forge" do
+    url "https://github.com/arrayfire/forge/archive/af3.0.tar.gz"
+    sha256 "c5664f272b5fdf5b76c782921d9f401578bd81b0534b4dfe710f4708198393bb"
+  end
+
+  patch do
+    url "https://github.com/glehmann/arrayfire/commit/6ed384c0d073c38d02e00e700a71ea6d85cde692.patch"
+    sha256 "032dc323c304d6b1c196f7779ddce691d08ae53baab97106da4ea74ae08b6293"
+  end
```
Comment (tdsmith) [inline arrayfire.rb]: Has this been submitted to the arrayfire devs?

File: arrayfire.rb [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
```
Comment (tdsmith) [inline arrayfire.rb]: It looks like there should be a `needs :cxx11` declaration here in the class body (and there should be a call to `ENV.cxx11` at the top of `install`)

File: arrayfire.rb (hunk 2 above)
Comment (glehmann) [inline arrayfire.rb]: yes, I've updated the link to use the pull request

File: arrayfire.rb (hunk 3 above)
Comment (glehmann) [inline arrayfire.rb]: thanks! it helps a lot :-)

There is still another problem though - see https://github.com/arrayfire/arrayfire/issues/794 . I'm going to exclude mountain lion for now.

File: boost-compute.rb (hunk 1 above)
Comment (glehmann) [inline boost-compute.rb]: I've updated the homepage. I think the tarball url is still ok though.

File: arrayfire.rb (hunk 2 above)
Comment (tdsmith) [inline arrayfire.rb]: :+1:

File: boost-compute.rb (hunk 1 above)
Comment (tdsmith) [inline boost-compute.rb]: It redirects but I think it's confusing.

**Comment by glehmann** [comment]
tests are ok, except on mountain lion where the audit fails. Is there is something wrong on the test host?

**Comment by tdsmith** [comment]
Some new strict audit tests were added yesterday and it looks like they're breaking on Ruby 1.8; I just let Homebrew/homebrew know.

**Comment by glehmann** [comment]
I don't get why there is “No test results found”. Any clue?

**Comment by tdsmith** [comment]
Sorry for the delay. There's some CI problem here:

```
==> brew uses Homebrew/science/arrayfire				 PASSED
Error: No available formula for glfw3 
Build step 'Execute shell' marked build as failure
```

**Comment by tdsmith** [comment]
Oh: that should be `depends_on "homebrew/versions/glfw3"`

**Comment by dpo** [comment]
@BrewTestBot test this please

**Comment by dpo** [comment]
@BrewTestBot test this please

**Comment by glehmann** [comment]
thanks!

**Pull request closed without merging**